src/codemodder/utils/utils.py

```diff
@@ -1,6 +1,14 @@
 import libcst as cst
 
 
+def list_subclasses(kls) -> set[str]:
```

Comment:
For `BaseException` this can be known up-front and will be constant for the entire process lifetime. This suggests we should either compute this up-front or do it lazily with a cache. You could use `@functools.lru_cache` but since this is called recursively it might be necessary to do a bit of refactoring.